doc/nrf/app_dev/device_guides/nrf54l/ecies_x25519.rst

```diff
@@ -0,0 +1,61 @@
+.. _ecies_x25519:
+
+MCUboot AES Image Encryption with ECIES-X25519 Key Exchange
+###########################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Introduction
+************
+
+MCUboot, when used with nRF54l15, can be configured to support encrypted images using AES.
+Images are encrypted using AES and ECIES-X25519 is used for key delivery (exchange) within image.
+When image encryption is enabled user may choose to either upload just signed or encrypted images to be swapped during boot.
+If MCUboot finds encrypted image in secondary slot it will decrypt it while swapping slots.
+Image that has been encrypted before being swapped into primary slot will be re-encrypted in case when swap is reverted.
+
+Limitations
+***********
+
+ECIES-X25519 key exchange is supported only when ED25519 signature is selected, which is default for nRF54l15.
+Encryption is not supported with MCUboot Direct-XIP modes.
+Storing ECIES-X25519 device private key in KMU is not supported at this moment.
+SHA256 is currently used by HMAC/HKDF, which is a subject to change.
```

Comment:
```suggestion
The current implementation has the following limitations:

* On the nRF54L15 SoC, ECIES-X25519 key exchange is supported only when the ED25519 signature algorithm is selected.
  This is the default configuration.
* Encryption is not supported when using MCUboot in Direct-XIP mode.
* Storing the ECIES-X25519 device private key in the Key Management Unit (KMU) is not supported at this time.
* SHA-256 is currently used by HMAC and HKDF, but this may change in future releases.
```